app/src/main/java/com/imdbmovieapp/data/repository/ApiMovieRepositoryImpl.kt

```diff
@@ -0,0 +1,75 @@
+package com.imdbmovieapp.data.repository
+
+import com.imdbmovieapp.data.remote.api.MoviesApi
+import com.imdbmovieapp.data.remote.mapper.dtoToDomain.DetailDtoToDetailDomainMapper
+import com.imdbmovieapp.data.remote.mapper.dtoToDomain.GenreDtoToGenreDomainMapper
+import com.imdbmovieapp.data.remote.mapper.dtoToDomain.PopularDtoToPopularDomainMapper
+import com.imdbmovieapp.data.remote.mapper.dtoToDomain.SearchDtoToSearchDomainMapper
+import com.imdbmovieapp.data.remote.mapper.dtoToDomain.TopRatedDtoToTopRatedDomainMapper
+import com.imdbmovieapp.domain.model.DetailMovieDomain
+import com.imdbmovieapp.domain.model.GenreMoviesDomain
+import com.imdbmovieapp.domain.model.PopularMoviesDomain
+import com.imdbmovieapp.domain.model.SearchMoviesDomain
+import com.imdbmovieapp.domain.model.TopRatedMoviesDomain
+import com.imdbmovieapp.domain.repository.ApiMovieRepository
+
+class ApiMovieRepositoryImpl(
+    private val moviesApi: MoviesApi,
+    private val popularDtoToPopularDomainMapper: PopularDtoToPopularDomainMapper,
+    private val topRatedDtoToTopRatedDomainMapper: TopRatedDtoToTopRatedDomainMapper,
+    private val detailDtoToDetailDomainMapper: DetailDtoToDetailDomainMapper,
+    private val searchDtoToSearchDomainMapper: SearchDtoToSearchDomainMapper,
+    private val genreDtoToDomainDto: GenreDtoToGenreDomainMapper
+) : ApiMovieRepository {
+    override suspend fun getPopularMovies(): List<PopularMoviesDomain> {
+        moviesApi.getPopularMovies().let { response ->
```

Comment:
there is no point to have .let without ? before (I mean ?.let) and when moviesApi doesn't return nullable, you don't have to use .let